consensus/spos/bls/subroundStartRound.go

```diff
@@ -186,17 +180,17 @@ func (sr *subroundStartRound) initCurrentRound() bool {
 
 	pubKeys := sr.ConsensusGroup()
 	numMultiKeysInConsensusGroup := sr.computeNumManagedKeysInConsensusGroup(pubKeys)
+	if numMultiKeysInConsensusGroup > 0 {
+		log.Debug("in consensus group with multi keys identities", "num", numMultiKeysInConsensusGroup)
+	}
 
 	sr.indexRoundIfNeeded(pubKeys)
 
-	isSingleKeyLeader := leader == sr.SelfPubKey() && sr.ShouldConsiderSelfKeyInConsensus()
-	isLeader := isSingleKeyLeader || sr.IsKeyManagedByCurrentNode([]byte(leader))
-	isSelfInConsensus := sr.IsNodeInConsensusGroup(sr.SelfPubKey()) || numMultiKeysInConsensusGroup > 0
-	if !isSelfInConsensus {
+	if !sr.IsSelfInConsensusGroup() {
```

Comment:
was this tested in the context of main machine and backup machine?
single and multikey
there are multiple combinations.

Reply:
no, this was not tested with a backup machine yet